lib/active_merchant/billing/credit_card_formatting.rb

```diff
@@ -15,6 +15,7 @@ def format(number, option)
         case option
         when :two_digits  then sprintf('%.2i', number.to_i)[-2..-1]
         when :four_digits then sprintf('%.4i', number.to_i)[-4..-1]
+        when :four_digits_year then number.to_s.length == 2 ? '20' + number.to_s : number.to_s
```

Comment:
The other two cases protect against a longer number being given; this new case will always yield 4 digits if the input is 2 digits, but for any other digit count, it will equal the input digit count.

I might recommend switching to leverage the existing four digit year code:

```suggestion
        when :four_digits_year then number.to_s.length == 2 ? '20' + number.to_s : format(number, :four_digits)
```

Reply:
That is a good suggestion, I will change to use that